core/server/master/src/main/java/alluxio/master/DefaultAlluxioMaster.java

```diff
@@ -262,33 +234,19 @@ public void stop() throws Exception {
   protected void startMasters(boolean isLeader) {
     try {
       connectToUFS();
-
-      mBlockMaster.start(isLeader);
-      mFileSystemMaster.start(isLeader);
-      if (LineageUtils.isLineageEnabled()) {
-        mLineageMaster.start(isLeader);
-      }
-      // start additional masters
-      for (Master master : mAdditionalMasters) {
+      for (Master master : mRegistry.getMasters()) {
```

Comment:
I also thought block master had to be started before the file master.

I think the master has to start to get all the journaling setup, and the file master will create the root inode, which will get an id from the block master, which will journal.